js/modules/k6/grpc/client.go

```diff
@@ -287,10 +299,65 @@ func (c *Client) Connect(ctxPtr *context.Context, addr string, params map[string
 	if err := <-errc; err != nil {
 		return false, err
 	}
-
+	if reflect {
+		var err error
+		c.sync.Do(func() {
+			err = c.reflect(ctxPtr)
+		})
+		if err != nil {
+			return false, fmt.Errorf("error invoking reflect API: %w", err)
+		}
```

Comment:
The `err` will only be populated on the first run which means that if you run multiple iterations and the first one fails reflect the API you will get 
```
ERRO[0000] error invoking reflect API: rpc error: code = Unimplemented desc = unknown service grpc.reflection.v1alpha.ServerReflection
        at go.k6.io/k6/js/common.Bind.func1 (native)
        at file:///home/mstoykov/work/k6io/k6/samples/grpc_reflection.js:7:64(9)  executor=shared-iterations scenario=default source=stacktrace
ERRO[0000] method "/main.RouteGuide/GetFeature" not found in file descriptors
        at go.k6.io/k6/js/common.Bind.func1 (native)
        at file:///home/mstoykov/work/k6io/k6/samples/grpc_reflection.js:9:13(21)  executor=shared-iterations scenario=default source=stacktrace
ERRO[0000] method "/main.RouteGuide/GetFeature" not found in file descriptors
        at go.k6.io/k6/js/common.Bind.func1 (native)
        at file:///home/mstoykov/work/k6io/k6/samples/grpc_reflection.js:9:13(21)  executor=shared-iterations scenario=default source=stacktrace
ERRO[0000] method "/main.RouteGuide/GetFeature" not found in file descriptors
        at go.k6.io/k6/js/common.Bind.func1 (native)
        at file:///home/mstoykov/work/k6io/k6/samples/grpc_reflection.js:9:13(21)  executor=shared-iterations scenario=default source=stacktrace
ERRO[0000] method "/main.RouteGuide/GetFeature" not found in file descriptors
        at go.k6.io/k6/js/common.Bind.func1 (native)
        at file:///home/mstoykov/work/k6io/k6/samples/grpc_reflection.js:9:13(21)  executor=shared-iterations scenario=default source=stacktrace
```

as you can see the much more prevalent message is that k6 couldn't find file descriptors.

This also means that if the connection fails for any reason on the first request you practically bricked this connection.

Additional problems:
- if the server reloads to change the definition and the user decided to use reflection so that ... works, it won't
- The connection can technically connect to multiple servers over many iterations, but this will prevent it to load different definitions for them.

I currently am of the opinion that it might be better off to *not* cache the definition and keep reloading them on each iteration just so that we don't tie our hands now. On the other hand, IMO this whole grpc API will be thrown out at some point when we start implementing everything it should support, so ... might as well have caching now :shrug: .

From my experiments. 100 VUs with caching it does 2.3k without caching it does a bit above 1k iter/s and with the `__ITER == 0` hack around 10k.

We can obviously keep adding flags, but again IMO the whole grpc module will need a rewrite and redoing of the API at some point so I am not certain how much adding flags for corner cases as the mentioned above is ... reasonable.

Reply:
>if the server reloads to change the definition and the user decided to use reflection so that ... works, it won't

is this a problem? grpc reflection isn't really a dynamic thing that is expected to change definitions ever, at least in all the time that I've used it